core/trino-main/src/main/java/io/trino/operator/project/PageFilter.java

```diff
@@ -29,9 +29,7 @@ static SelectedPositions positionsArrayToSelectedPositions(boolean[] selectedPos
         int selectedCount = 0;
         for (int i = 0; i < size; i++) {
             boolean selectedPosition = selectedPositions[i];
-            if (selectedPosition) {
-                selectedCount++;
-            }
+            selectedCount += selectedPosition ? 1 : 0;
```

Comment:
Current code is not super readable and also looks like it's doing more operations (always boolean test and always `+=`, while previous code was doing a boolean test and increment sometimes).

Add a comment with the intent, so it doesn't get  "improved back".